staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1365,9 +1367,36 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 		)
 	}
 
+	if opts.CountMetricPollPeriod > 0 {
+		stopFunc := e.startObservingCount(opts.CountMetricPollPeriod)
+		previousDestroy := e.DestroyFunc
+		e.DestroyFunc = func() {
+			stopFunc()
+			if previousDestroy != nil {
+				previousDestroy()
+			}
+		}
+	}
+
 	return nil
 }
 
+// startObservingCount starts monitoring given prefix and periodically updating metrics. It return a function to stop collection.
+func (e *Store) startObservingCount(period time.Duration) func() {
+	prefix := e.KeyRootFunc(genericapirequest.NewContext())
+	resourceName := e.DefaultQualifiedResource.String()
+	glog.V(2).Infof("Monitoring %v bound to <storage-prefix>/%v", resourceName, prefix)
+	stopCh := make(chan struct{})
+	go wait.Until(func() {
```

Comment:
Do we really want a separate go routine for every resource type? As opposed to a single routine that loops through all resources (possibly waiting between each)?

Reply:
That's the cost of doing it at the store interface level. Alternative would be to "register" types here for the Storage to use, but it would required leaking stuff from the higher to lower levels of the stack. I didn't like this design.

Or we could have a global registry of things to pull, but I really don't like global variables.

Functionally it's not a big difference (except the fact that polling in completely unsynchronized way can cause few calls to be issued in the same-ish moment). It it proves to be an issue, it's easy enough to change to global variable approach.